### ISSUE TITLE
Peer DID: Backing Storage

### ISSUE THREAD

**Issue by rolsonquadras** [hyperledger/aries-framework-go#47]
The agent should provide a [storage](https://openssi.github.io/peer-did-method-spec/index.html#backing-storage) to save the Peer DID Docs. 

The peer DID docs need to be associated with some type of backing storage that adds metadata and history (not raw documents). It should store a sequence of _deltas_ and each delta is a JSON object.

Delta structure:
```

{ "change": <base64 encoding of a change fragment>, "by": [ 

{"key": <id of key>, "sig": <signature value>}

... ],
"when": <ISO8601 UTC timestamp with at least second precision>
}
```

Here, a _change fragment_ is a sparse version of a DID doc that shows just the section being updated. Also, the agent should have the capability to construct a Peer DID Document from these deltas.

**Comment by rolsonquadras** [comment]
Part of epic #35

**Comment by llorllale** [comment]
@rolsonquadras for iteration 0 we definitely need storage of DIDs, but we don't anticipate any updates (and conflict resolution)

**Comment by rolsonquadras** [comment]
Created #54 for conflict resolution requirement and updated the title of this story.

**Comment by rolsonquadras** [comment]
The changes are implemented in #55
